app/src/PageController.php

```diff
@@ -3,9 +3,45 @@
 namespace {
 
     use SilverStripe\CMS\Controllers\ContentController;
+    use SilverStripe\Control\Cookie;
+    use SilverStripe\Control\Director;
+    use SilverStripe\Control\Session;
+    use SilverStripe\Core\Environment;
+    use SilverStripe\View\ArrayData;
+    use SilverStripe\View\Requirements;
 
     class PageController extends ContentController
     {
 
+        /**
+         * Whether our modal should be showed on this request.
+         * @var bool
+         */
+        private $showModal = false;
+
+        protected function init()
+        {
+            parent::init();
+
+            // Inject our privacy modal assets into the page
+            $cookie = Cookie::get('privacy_modal_showed');
+            if (empty($cookie)) {
```

Comment:
We'll probably add HTTP caching to this, so I'd prefer if that conditional was handled on the client, rather than creating uncacheable responses.

Reply:
It's gone, all client side now ... FYI this means the modal and related style and JS will be served on every page.